assets/js/app/television/television.controller.js

```diff
@@ -0,0 +1,286 @@
+ (function () {
+    'use strict';
+
+    angular
+        .module('gladys')
+        .controller('TelevisionCtrl', TelevisionCtrl);
+
+    TelevisionCtrl.$inject = ['televisionService', 'deviceService', 'boxService'];
+
+    function TelevisionCtrl(televisionService, deviceService, boxService) {
+        /* jshint validthis: true */
+        var vm = this;
+        vm.devices = [];
+        vm.devicetypes = [];
+        vm.deviceId = null;
+        vm.displayAskDeviceForm = false;
+        vm.currentPowerState = '';
+        vm.currentSoundState = '';
+        vm.currentChannel = '';
+        vm.currentMuteState = '';
+        vm.defaultSource = [{ label: 'TV' }];
+        vm.allSources = vm.defaultSource;
+        vm.currentSource = vm.defaultSource[0].label;
+        vm.currentDeviceName = '';
+
+        vm.selectDevice = selectDevice;
+
+        vm.play = play;
+        vm.pause = pause;
+        vm.stop = stop;
+        vm.rewind = rewind;
+        vm.fastForward = fastForward;
+        vm.switchState = switchState;
+        vm.getState = getState;
+        vm.programPlus = programPlus;
+        vm.programMinus = programMinus;
+        vm.setChannel = setChannel;
+        vm.setMuted = setMuted;
+        vm.volumeUp = volumeUp;
+        vm.volumeDown = volumeDown;
+        vm.pressKey = pressKey;
+        vm.getSources = getSources;
+        vm.openSource = openSource;
+        vm.openMenu = openMenu;
+        vm.rec = rec;
+        vm.customCommand = customCommand;
+
+        vm.thisChannel = null;
+
+        vm.init = init;
+
+        getDevices();
+
+        function init(id) {
+            vm.boxId = id;
+            
+            boxService.getById(id)
+                .then(function (data) {
+                    vm.box = data.data;
+                    if (vm.box.params && vm.box.params.deviceId) {
+                        setBoxInformation(vm.box.params.deviceId, vm.box.params.name)
+                    } else {
+                        vm.displayAskDeviceForm = true;
+                    }
+                });
+        }
+
+        function setBoxInformation(deviceId, deviceName){
+            vm.displayAskDeviceForm = false;
+            vm.deviceId = deviceId;
+            vm.currentDeviceName = deviceName;
+            vm.currentPowerState = null;
+            vm.currentSoundState = null;
+            vm.currentChannel = null;
+            vm.currentMuteState = null;
+            getData(deviceId);
+            getSources();
+        }
+
+        function getDevices() {
+            deviceService.getDeviceTypeByCategory({category:'TV'})
```

Comment:
Just so you know, as here the feature is now called "television", I'm going to rename the category to "television" as well so it's more consistent.

I'll change the get here while merging! Any other place to change that?

Reply:
Changing on commit 6f08686